src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -837,6 +841,25 @@ private bool TryComputeWeightedDistance(BlockSyntax leftBlock, BlockSyntax right
             }
         }
 
+        private double ComputeWeightedDistance(SingleVariableDesignationSyntax leftNode, SingleVariableDesignationSyntax rightNode)
+        {
+            double distance = ComputeDistance(leftNode, rightNode);
+            double parentDistance;
+
+            if (leftNode.Parent != null && 
+                rightNode.Parent != null &&
+                GetLabel(leftNode.Parent) == GetLabel(rightNode.Parent))
+            {
+                parentDistance = ComputeDistance(leftNode.Parent, rightNode.Parent);
```

Comment:
I think this is a good idea, but wonder if we need to tailor it a bit more to switch case.

In future C# will support complex pattern matching in switch, so the direct parent of a variable designation won't be the case label. 

How about find the first containing node that has a label instead of the direct parent?